### PR TITLE
Remove use of deprecated function sdf::JointAxis::InitialPosition

### PR DESCRIPTION
**Pull request by azeey**
## Summary
https://github.com/ignitionrobotics/sdformat/pull/622 will remove `sdf::JointAxis::InitialPosition`, so we need to stop using it here.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: dartsim/src/SDFFeatures.cc [hunk 1]
```diff
@@ -1115,30 +1101,6 @@ Identity SDFFeatures::ConstructSdfJoint(
   joint->setTransformFromParentBodyNode(parent_T_prejoint_init);
   joint->setTransformFromChildBodyNode(child_T_postjoint);
```
Comment (chapulina) [inline dartsim/src/SDFFeatures.cc:1102]: Is this block up here still needed? The comment seems outdated

**Comment by darksylinc** [comment]
I am trying to build Fortress and hit this issue. InitialPosition has been removed so ign-physics doesn't build out of the box.

File: dartsim/src/SDFFeatures.cc (hunk 1 above)
Comment (scpeters) [reply dartsim/src/SDFFeatures.cc:1102]: agreed, the comment seems outdated

**Comment by scpeters** [comment]
the ign-physics nightly build is broken until we get this in

File: dartsim/src/SDFFeatures.cc (hunk 1 above)
Comment (scpeters) [reply dartsim/src/SDFFeatures.cc:1102]: I'm going to merge this now to unbreak things but will open an issue to follow up on this comment

**Review by scpeters — APPROVED** [review]

File: dartsim/src/SDFFeatures.cc (hunk 1 above)
Comment (azeey) [reply dartsim/src/SDFFeatures.cc:1102]: Yes, the code is needed, but the comment is indeed outdated. I think it can simply be removed.

**Comment by azeey** [comment]
My bad for merging https://github.com/ignitionrobotics/sdformat/pull/622 before merging this.

File: dartsim/src/SDFFeatures.cc (hunk 1 above)
Comment (azeey) [reply dartsim/src/SDFFeatures.cc:1102]: https://github.com/ignitionrobotics/ign-physics/pull/280

**Comment by scpeters** [comment]
> My bad for merging [ignitionrobotics/sdformat#622](https://github.com/ignitionrobotics/sdformat/pull/622) before merging this.

no worries, it was easy to fix

**Pull request merged**